src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
+
+    private int hash(K key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private void add(K key, V value) {
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int number = ((table.length - 2) & hash(key)) + 1;
```

Comment:
```suggestion
        int index = getIndex(hash);
```
lets create a separate method to calculate index, this method should also calculate index for null key